util/patch/patch.go

```diff
@@ -124,7 +128,7 @@ func (h *Helper) Patch(ctx context.Context, obj client.Object, opts ...Option) e
 	}
 
 	// Calculate and store the top-level field changes (e.g. "metadata", "spec", "status") we have before/after.
-	h.changes, err = h.calculateChanges(obj)
+	h.changes, err = h.calculateChanges(ctx, obj, options)
 	if err != nil {
```

Comment:
I would prefer adding the log on this level ~ l.135 instead of making it a part of Calculating changes. It just feels counter-intuitive to me that calculate changes (already) logs that we are patching something.